### ISSUE TITLE
REST API 5.11 Support For WooCommerce Products, Orders, Coupons

### ISSUE THREAD

**Issue by xHeinrich** [AdvancedCustomFields/acf#580]
Add ability to customize supported routes, object_type, and object_child_type in the ACF_Rest_Request.

Currently, there is no out-of-the-box support for updating custom fields via the REST API on WooCommerce custom post types.

Would be good if these are either supported out of the box or have the ability to update the ACF_Rest_Request supported routes as well as object_type and object_child_type via filters so that we can do this ourselves.

Example code that I've whipped up that adds support via filters.

Line 145: add ability to update supported routes
`        $this->supported_routes = apply_filters('acf/rest_request_supported_routes', $this->supported_routes);
`

Line 206: add ability to update object_type and object_sub_type
`        $this->object_type = apply_filters('acf/rest_request_object_type', $this->object_type, $base, $child_base);
        $this->object_sub_type = apply_filters('acf/rest_request_child_object_type', $this->object_type, $base, $child_base);
`

Example filters to add support for WooCommerce Products
`
add_filter('acf/rest_request_supported_routes', function($supported_routes) {
  $supported_routes[] = '/wc/v3/(?P<rest_base>products)/(?P<id>[\d]+)';
  return $supported_routes;
});

add_filter('acf/rest_child_object_type', function($object_type, $base) {
  if(!$object_type && $base === 'products') {
    return 'products';
  }
  return $object_type;
}, 2, 10);

add_filter('acf/rest_request_object_type', function($object_type, $base) {
  if(!$object_type && $base === 'products') {
    return 'products';
  }
  return $object_type;
}, 2, 10);

`

**Comment by jackgllghr** [comment]
Agreed, I'm currently using a second plugin to get support this but would be good if it was all handled in this. This supports it out of the box:
https://github.com/airesvsg/acf-to-rest-api
